### DISCUSSION TITLE
Unsupported class file major version 63

### DISCUSSION THREAD

**Discussion by MarcusSlover** [FabricMC/fabric-api#3082] [Mod Dev Support] [answered]
I'm trying to make a 1.19.4 Mod using Kotlin with the latest loom and Java 17.
The error message mentions `Version 63` which is Java 19... what is happening?

Error:
https://pastebin.com/dXxB9N3p

No idea how to fix it.
Here's my grade (it's a sub-project):
```kotlin
plugins {
    id("fabric-loom") version "1.2-SNAPSHOT"
}

repositories {
    maven { url = uri("https://maven.fabricmc.net/") }
}

dependencies {
    minecraft("com.mojang:minecraft:1.19.4")
    mappings("net.fabricmc:yarn:1.19.4+build.2")
    modImplementation("net.fabricmc:fabric-loader:0.14.21")

    val apiModules = arrayOf(
        "fabric-api-base",
        "fabric-command-api-v2",
        "fabric-lifecycle-events-v1",
        "fabric-networking-api-v1"
    )

    apiModules.forEach {
        modImplementation(fabricApi.module(it, "0.82.0+1.19.4"))
    }
    implementation(project(":case-common"))
}

tasks {
    shadowJar {
        dependencies {
            exclude("net.fabricmc:.*")
            exclude("/mappings/*")
        }
    }
}

```

My main-main build.gradle,kts
```kotlin
plugins {
    kotlin("jvm") version "1.8.21"
    id("com.github.johnrengelman.shadow") version "7.1.2"
    `maven-publish`
    java
}

allprojects {
    group = "com.marcusslover.case"
    version = "1.0-SNAPSHOT"

    apply(plugin = "kotlin")
    apply(plugin = "com.github.johnrengelman.shadow")
    apply(plugin = "maven-publish")

    repositories {
        mavenCentral()
    }

    dependencies {
        testImplementation(kotlin("test"))
    }

    tasks.test {
        useJUnitPlatform()
    }

    kotlin {
        jvmToolchain(17)
    }

    java {
        withSourcesJar()
    }
}
 ```
 
 If you have any clue what I may be doing wrong, please let me know.
 I tried deleting the loom caches in Gradle, even getting a new manifest file https://piston-meta.mojang.com/mc/game/version_manifest.json

**Comment by Juuxel** [comment, marked as answer]
The Java 19 class is contained within LWJGL:
> error analyzing /META-INF/versions/19/org/lwjgl/panama/Test.class from C:\Users\MarcusSlover\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl\3.3.1\ae58664f88e18a9bb2c77b063833ca7aaec484cb\lwjgl-3.3.1.jar

I think what's happening is that the Gradle runtime classpath contains an outdated version of ASM since you don't specify a Loom dependency at the root project level.

This should fix any classpath issues:
1. add `id("fabric-loom") version "1.2-SNAPSHOT" apply false` to your `plugins {}` in the root build.gradle.kts
2. use `id("fabric-loom")` (without the version) in the Fabric build.gradle.kts

(As a side note, you don't need to declare the Fabric maven when you're using Loom since it gets declared automatically.)

**Reply by MarcusSlover** [reply]
Thank you @Juuxel for helping, it worked!
